src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -213,21 +244,72 @@ private boolean checkInstallationDir(Path path) throws NoSuchFileException, Acce
+            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTPS_PASSWORD));
+        } else if (httpHost != null && StringUtils.isNotBlank(httpHost)) {
+            logger.warn("{} Use of HTTP instead of HTTPS for Proxy system environment properties was found. "
+                    + "It is strongly recommended to use HTTPS over HTTP for increased security."
+                    + "Detect will continue self update check with the configured HTTP settings.");
+            proxyProperties.put(ARG_PROXY_PORT, httpHost);
+            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTP_PORT));
+            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTP_USERNAME));
+            proxyProperties.put(ARG_PROXY_PORT, System.getenv(SYS_ENV_PROP_PROXY_HTTP_PASSWORD));
+        }
+    }
+    
     private String[] parseArguments(String[] args) {
+        final Map<String, String> tempProxyProperties = new HashMap<>(7);
         final ListIterator<String> it = Arrays.asList(args).listIterator();
         while (it.hasNext()) {
             String argument = it.next();
-            if (argument.contains(BLACKDUCK_OFFLINE_MODE_ARG)) {
-                offlineMode = findArgument(it, argument);
-            } else if (argument.contains(BLACKDUCK_URL_ARG)) {
-                blackduckHost = findArgument(it, argument);
+            if (argument.contains(ARG_BLACKDUCK_OFFLINE_MODE)) {
+                offlineMode = findArgumentValue(it, argument);
+            } else if (argument.contains(ARG_BLACKDUCK_URL)) {
+                blackduckHost = findArgumentValue(it, argument);
+            } else if (argument.contains(ARG_TRUST_CERTIFICATE)) {
+                trustCertificate = findArgumentValue(it, argument).toLowerCase().equals("true");
+            } else if (argument.contains(ARG_PROXY_HOST)) {
+                addProxyPropertyToTempMap(ARG_PROXY_HOST, it, argument, tempProxyProperties);
+            } else if (argument.contains(ARG_PROXY_PORT)) {
+                addProxyPropertyToTempMap(ARG_PROXY_PORT, it, argument, tempProxyProperties);
+            } else if (argument.contains(ARG_PROXY_PASSWORD)) {
+                addProxyPropertyToTempMap(ARG_PROXY_PASSWORD, it, argument, tempProxyProperties);
+            } else if (argument.contains(ARG_PROXY_USERNAME)) {
+                addProxyPropertyToTempMap(ARG_PROXY_USERNAME, it, argument, tempProxyProperties);
+            } else if (argument.contains(ARG_PROXY_NTLM_DOMAIN)) {
+                addProxyPropertyToTempMap(ARG_PROXY_NTLM_DOMAIN, it, argument, tempProxyProperties);
+            } else if (argument.contains(ARG_PROXY_NTLM_WORKSTATION)) {
+                addProxyPropertyToTempMap(ARG_PROXY_NTLM_WORKSTATION, it, argument, tempProxyProperties);
+            } else if (argument.contains(ARG_PROXY_IGNORED_HOSTS)) {
+                proxyIgnoredHosts = findArgumentCommaDelimitedValues(it, argument);
+            }
```

Comment:
Any proxy setting arguments found are loaded into a temporary map so that we can discard them entirely if the proxy host and port are not set in the arguments. This was, incomplete proxy argument settings do not overwrite complete proxy settings set through system environment properties.